lib/viking/record/associations/collectionAssociation.js

```diff
@@ -255,26 +256,19 @@ export default class CollectionAssociation extends Association {
             return this.owner.constructor.primaryKey;
         }
     }
-    
-    saveAttributes (options) {
-        let record_attributes = this.target.map(record => {
-            const attributes = record.saveAttributes(options)
-            if (Object.keys(attributes).length == 0) {
-                return null
-            }
+
+    autosave() {
+        return (this.dirty || this.target.some(r => r.hasChanged()));
```

Comment:
```suggestion
        return (this.dirty || this.target.some(r => r.needsSaved()));
```
Gotta check records sub resources